common/src/main/java/com/wynntils/wynn/model/SpellModel.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.wynn.event.SpellCastedEvent;
+import com.wynntils.wynn.model.actionbar.event.SpellSegmentUpdateEvent;
+import com.wynntils.wynn.objects.SpellType;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class SpellModel extends Model {
```

Comment:
There seems to be opportunities here to move functionality from QuickCastFeature to this new Model. But maybe that is a separate follow-up project?

Reply:
I honestly thought this was straight up copied from there?